src/tests/ftest/util/data_mover_test_base.py

```diff
@@ -288,7 +247,8 @@ def new_posix_test_path(self, shared=False, create=True, parent=None, mount_dir_
             cmd = f"sudo mount -t tmpfs none '{path}' -o size={mount_dir_size}"
             if not run_remote(self.log, self.hostlist_clients, cmd).passed:
                 self.fail(f"Failed to mount directory {path}")
-            self.mounted_posix_test_paths.append(path)
+            self.register_cleanup(
+                cleanup_mounted_path, test=self, hosts=self.hostlist_clients, path=path)
```

Comment:
We need to do this before removing the directory

Reply:
Sorry, I read this backward

Reply:
I was looking into why this failed
https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-14245/7/artifact/Functional%20on%20EL%208.8/datamover/negative_space.py/job.log
```
2024-05-16 00:18:14,546 test             L0411 DEBUG| [7] Register: Calling cleanup method cleanup_mounted_path(test=1-./datamover/negative_space.py:DmvrNegativeSpaceTest.test_dm_negative_space_dcp;run-container-dcp-client_processes-hosts-ior-pool_large-pool_small-server_config-engines-0-storage-0-981d, hosts=wolf-105vm3, path=/mnt/share/tmp.2hS8f75viA/test_dm_negative_space_dcp0)
2024-05-16 00:18:14,546 run_utils        L0410 DEBUG| Running on wolf-105vm3 with a 120 second timeout: rm -rf '/mnt/share/tmp.2hS8f75viA/test_dm_negative_space_dcp0'
2024-05-16 00:18:14,925 run_utils        L0264 DEBUG|   wolf-105vm3 (rc=1): rm: cannot remove '/mnt/share/tmp.2hS8f75viA/test_dm_negative_space_dcp0': Device or resource busy
2024-05-16 00:18:14,926 run_utils        L0410 DEBUG| Running on wolf-105vm3 with a 120 second timeout: sudo umount -f '/mnt/share/tmp.2hS8f75viA/test_dm_negative_space_dcp0'

```